### DISCUSSION TITLE
Understanding the Blocks Indexed Calculation and Speed Optimization

### DISCUSSION THREAD

**Discussion by iw4p** [blockscout/blockscout#8881] [Q&A]
Hello,

I am currently trying to grasp the concept and mechanics behind the calculation of 'Blocks Indexed' in a blockchain context. I've noticed that the progress appears to be quite slow. For instance, a persistent banner on the website indicates "11% Blocks Indexed - We're indexing this chain right now. Some of the counts may be inaccurate."

<img width="548" alt="Screenshot 2023-11-24 at 3 16 07 PM" src="https://github.com/blockscout/blockscout/assets/30632761/1fb170b7-55f1-4999-8a1c-83aa63d897de">

Could someone provide a detailed explanation on how 'Blocks Indexed' is calculated? Additionally, are there any strategies or best practices that can be employed to accelerate the indexing process? I'm particularly interested in any settings or optimizations that could enhance the speed of this process.

Any insights or guidance on this matter would be greatly appreciated.

Thank you.

**Comment by vbaranov** [comment]
The indexing percentage in the alert re-calculates every 2hrs. If you want to invalidate the cache frequentlier, you could set `CACHE_BLOCK_COUNT_PERIOD` to the lower value. Say, `CACHE_BLOCK_COUNT_PERIOD=10m`. Then update will take in place every 10 minutes.
